tests/e2e/resources/installation_config/cognito.yaml

```diff
@@ -0,0 +1,272 @@
+#level 1
+#cert-manager
+cert-manager: 
+  installation_options:  
+    kustomize: 
+      - "../../upstream/common/cert-manager/cert-manager/base"
+    helm: "remote repo"
+  validations:
+    namespace: 'cert-manager'
+    identifier: 'app.kubernetes.io/instance'
+    common_label: 'cert-manager'
+
+#kubeflow-roles
+kubeflow-roles:
+  installation_options:
+    kustomize: 
+      - "../../upstream/common/kubeflow-roles/base"
+    helm: "../../charts/common/kubeflow-roles"
+  
+#level 2
+#kubeflow-issuer
+kubeflow-issuer:
+  installation_options:
+    kustomize: 
+      - "../../upstream/common/cert-manager/kubeflow-issuer/base"
+    helm: "../../charts/common/kubeflow-issuer"
+
+#istio
+istio-1-14:
+  installation_options:
+    kustomize: 
+        - "../../upstream/common/istio-1-14/istio-crds/base"
+        - "../../upstream/common/istio-1-14/istio-namespace/base"
+        - "../../upstream/common/istio-1-14/istio-install/base"
+    helm: "../../charts/common/istio-1-14"
+  validations: 
+    namespace: 'istio-system'
+    identifier: 'app'
+    common_label: 'istio-ingressgateway, istiod'
+
+#kubeflow-namespace
+kubeflow-namespace:
+  installation_options:
+    kustomize: 
+      - "../../upstream/common/kubeflow-namespace/base"
+    helm: "../../charts/common/kubeflow-namespace"
+
+#cluster-local-gateway
+cluster-local-gateway:
+  installation_options:
+    kustomize:
+      - "../../upstream/common/istio-1-14/cluster-local-gateway/base"
+    helm: "../../charts/common/cluster-local-gateway"
+  validations: 
+    namespace: 'istio-system'
+    identifier: 'app'
+    common_label: 'cluster-local-gateway'
+
+#knative
+knative-serving:
+  installation_options:
+    kustomize: 
+      - "../../upstream/common/knative/knative-serving/overlays/gateways"
+    helm: "../../charts/common/knative-serving"
+  crd_required: 'images.caching.internal.knative.dev'
```

Comment:
this is also a validation. can we do:
```
validation:
  crds:
  pods:
```

Reply:
For me they were separate. crd_required is needed to install the component fully, that's why I call it crd_required as it's a prerequisite. do you want to do
```
validations: 
    crd: 
    identifier: 
    common_label:
    namespace: 
```
or
```
validations: 
    crd: 
    pods:
        identifier: 
        common_label:
        namespace: 
```

Reply:
```
validations:
  crds:
   - crd1
   - crd2
  pods:
   namespace:
   labels:
    - key:
      value:
```

if crds exist in validation dictionary the behaviour will be same as now
install component -> verify crds established -> reinstall component -> verify pods running with label1, verify pods running with label2...

Reply:
addressed

Reply:
I think the problem is: the crds established can only be done after some installation/resources are created. We are mixing up the two